### ISSUE TITLE
[FEATURE] <Generate a username when a user signs up>

### ISSUE THREAD

**Issue by ykdojo** [ykdojo/defaang#95]
### Description

We should generate a username when a user signs up, and we should have an option for them to change it.

**Comment by yung-coder** [comment]
Yeah, I think we can use an npm package unique-username-generator that can generate a username from email by adding some random digits or totally generate a random username ......

**Comment by AdityaPainuli** [comment]
Picking up a unique username is all good just a suggestion we can also use a random avatar as their profile pic. If it sounds good to all of you.

**Comment by yung-coder** [comment]
The avatar thing is great and should be put on consideration....

**Comment by iShibi** [comment]
I suggest we let the user pick a `username` during Registration. This will take the burden of generating random usernames away from us. If a user wants to change their username, later on, we should have an option to change it in settings.

For generating random avatars we can use the [Dicebear](https://github.com/dicebear/dicebear) library. It can generate different types of avatars using a seed. We can use the `username` as a seed for generating a user's avatar with it.

**Comment by AdityaPainuli** [comment]
Is there anyone working on this feature?

**Comment by ykdojo** [comment]
@AdityaPainuli no I don't think so

**Comment by omkarg01** [comment]
Hey @ykdojo, I would like to work on it.

**Comment by iShibi** [comment]
We have moved on from the initial username generation plan. Now, we let the user choose one for themselves. Closing this as it is no longer on the roadmap. I'll open another issue that addresses changing of username after #151.
